components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java

```diff
@@ -354,6 +355,162 @@ public void testAttachAnnotation() throws Exception {
+        PlateData p = new PlateData();
+        p.setName(UUID.randomUUID().toString());
+        sources.add(datamanagerFacility.saveAndReturnObject(rootCtx, p));
+        long imgId = createImage(rootCtx);
+        List<Long> ids = new ArrayList<Long>(1);
+        ids.add(imgId);
+        sources.add(browseFacility.getImages(rootCtx, ids).iterator().next());
+
+        List<DataObject> targets = new ArrayList<DataObject>();
+        ds = new DatasetData();
+        ds.setName(UUID.randomUUID().toString());
+        targets.add(datamanagerFacility.saveAndReturnObject(rootCtx, ds));
+        proj = new ProjectData();
+        proj.setName(UUID.randomUUID().toString());
+        targets.add(datamanagerFacility.saveAndReturnObject(rootCtx, proj));
+        s = new ScreenData();
+        s.setName(UUID.randomUUID().toString());
+        targets.add(datamanagerFacility.saveAndReturnObject(rootCtx, s));
+        p = new PlateData();
+        p.setName(UUID.randomUUID().toString());
+        targets.add(datamanagerFacility.saveAndReturnObject(rootCtx, p));
+        imgId = createImage(rootCtx);
+        ids = new ArrayList<Long>(1);
+        ids.add(imgId);
+        targets.add(browseFacility.getImages(rootCtx, ids).iterator().next());
+
+        Collection<AnnotationData> annos = new ArrayList<AnnotationData>();
+        annos.add(new BooleanAnnotationData(true));
+        annos.add(new DoubleAnnotationData(5d));
+        annos.add(new LongAnnotationData(1));
+        annos.add(new MapAnnotationData());
+        annos.add(new RatingAnnotationData(3));
+        annos.add(new TagAnnotationData("test"));
+        annos.add(new TermAnnotationData("test2"));
+        annos.add(new TextualAnnotationData("test3"));
+        annos.add(new XMLAnnotationData("<test4/>"));
+
+        for (int i = 0; i < sources.size(); i++) {
+            DataObject src = sources.get(i);
+            DataObject tar = targets.get(i);
+            List<AnnotationData> attached = new ArrayList<AnnotationData>();
+            annos.forEach(anno -> {
+                try {
+                    attached.add(datamanagerFacility.attachAnnotation(rootCtx, anno, src));
+                } catch (DSOutOfServiceException | DSAccessException e) {
+                    e.printStackTrace();
+                }
+            });
+            
+            datamanagerFacility.move(rootCtx, attached, src, tar);
+            List<AnnotationData> loaded = metadataFacility.getAnnotations(rootCtx, tar);
+            
+            Assert.assertEquals(attached.size(), loaded.size());
+            Set<Long> check = attached.stream().map(a -> a.getId()).collect(Collectors.toSet());
+            loaded.forEach(a -> Assert.assertTrue(check.contains(a.getId())));
+            
+            List<IObject> del = attached.stream().map(obj -> obj.asIObject()).collect(Collectors.toList());
+            datamanagerFacility.delete(rootCtx, del);
+            Thread.sleep(100);
```

Comment:
why do you need it?

Reply:
As far as I remember I randomly hit some errors if I didn't wait there for the deletion to be finished. I'll test again with that line.